src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Sdk.Before.targets

```diff
@@ -21,4 +23,41 @@
+    <_MauiPreviousTargetPlatformMinVersion>$(TargetPlatformMinVersion)</_MauiPreviousTargetPlatformMinVersion>
+    <!-- Windows has 2 properties that could be set, so if only 1 is set we copy it to the other. -->
+    <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' and '$(TargetPlatformMinVersion)' == ''">$(SupportedOSPlatformVersion)</TargetPlatformMinVersion>
+    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' and '$(SupportedOSPlatformVersion)' == ''">$(TargetPlatformMinVersion)</SupportedOSPlatformVersion>
+    <!-- Record that the explicit version number was lower than we support so we can increase and log later. -->
+    <_MauiHadToIncreaseSupportedOSPlatformVersion Condition="'$(SupportedOSPlatformVersion)' != '' and $([MSBuild]::VersionGreaterThan('$(_MauiSupportedOSPlatformVersion)', '$(SupportedOSPlatformVersion)'))">true</_MauiHadToIncreaseSupportedOSPlatformVersion>
+    <_MauiHadToIncreaseTargetPlatformMinVersion Condition="'$(TargetPlatformMinVersion)' != '' and $([MSBuild]::VersionGreaterThan('$(_MauiSupportedOSPlatformVersion)', '$(TargetPlatformMinVersion)'))">true</_MauiHadToIncreaseTargetPlatformMinVersion>
+    <!-- Specify the version numbers, and override if it is too low. -->
+    <SupportedOSPlatformVersion Condition="'$(SupportedOSPlatformVersion)' == '' or '$(_MauiHadToIncreaseSupportedOSPlatformVersion)' == 'true'">$(_MauiSupportedOSPlatformVersion)</SupportedOSPlatformVersion>
+    <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' and ('$(TargetPlatformMinVersion)' == '' or '$(_MauiHadToIncreaseTargetPlatformMinVersion)' == 'true')">$(SupportedOSPlatformVersion)</TargetPlatformMinVersion>
+  </PropertyGroup>
+
+  <!--
+    Log a warning if the user specified a version that was lower than the minimum supported version.
+  -->
+  <Target
+    Name="_CheckForMauiHadToIncreaseVersions"
+    BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+    Condition="'$(TargetFramework)' != '' and '$(MauiDisableSupportedOSPlatformVersionValidation)' != 'true'">
+    <Warning
+      Code="MA003"
+      Condition="'$(_MauiHadToIncreaseSupportedOSPlatformVersion)' == 'true'"
+      Text="The specified &lt;SupportedOSPlatformVersion&gt; of '$(_MauiPreviousSupportedOSPlatformVersion)' was older than the minimum supported version by .NET MAUI and so it was increased to '$(_MauiSupportedOSPlatformVersion)'. " />
+    <Warning
+      Code="MA003"
+      Condition="'$(_MauiHadToIncreaseTargetPlatformMinVersion)' == 'true'"
+      Text="The specified &lt;TargetPlatformMinVersion&gt; of '$(_MauiPreviousTargetPlatformMinVersion)' was older than the minimum supported version by .NET MAUI and so it was increased to '$(_MauiSupportedOSPlatformVersion)'. " />
+  </Target>
```

Comment:
Can we remove these warnings/checks from Android?

Reply:
If the version number is at the android min (21.0) then we take a mini-risk and say no warning, just a message for our binlogs. Tl;DR: no warnings for android.

Reply:
I am ok for the Android logic to be here, as long as there is nothing trying to automatically increase the `$(SupportedOSPlatformVersion)` number.